intro-HPC/ch_running_batch_jobs.tex

```diff
@@ -40,6 +40,331 @@ \chapter{Running batch jobs}
 of the general features of job scripts, how to submit them for execution and
 how to monitor their progress.
 
+\section{Modules}
+\label{sec:modules}
+
+Software installation and maintenance on a \hpc cluster such as the VSC
+clusters poses a number of challenges not encountered on a workstation or a
+departmental cluster. We therefore need a system on the \hpc, which is able
+to easily activate or deactivate the software packages that you require for
+your program execution.
+
+\subsection{Environment Variables}
+
+The program environment on the \hpc is controlled by pre-defined settings,
+which are stored in environment (or shell) variables.
+
+You can use shell variables to store data, set configuration options and
+customise the environment on the \hpc. The default shell
+on the \hpc is Bash (Bourne Again Shell) and can be used for the
+following purposes:
+
+\begin{enumerate}
+\item  Configure look and feel of the shell.
+\item  Setup terminal settings depending on which terminal you're using.
+\item  Set the search path for running executables.
+\item  Set environment variables as needed by programs.
+\item  Set convenient abbreviations for heavily used values.
+\item  Run commands that you want to run whenever you log in or log out.
+\item  Setup aliases and/or shell function to automate tasks to save typing and time.
+\item  Changing the bash prompt.
+\item  Setting shell options.
+\end{enumerate}
+
+The environment variables are typically set at login by a script, whenever you
+connect to the \hpc. These pre-defined variables usually impact the run time
+behaviour of the programs that we want to run.
+
+All the software packages that are installed on the \hpc cluster require
+different settings. These packages include compilers, interpreters,
+mathematical software such as MATLAB and SAS, as well as other applications and
+libraries.
+
+In order to administer the active software and their environment variables, a
+\strong{\emph{``module''}} package has been developed, which:
```

Comment:
the tool/concept is actually called "environment modules" (cfr. http://modules.sourceforge.net/, but don't include that link since some sites using a different "fork", Lmod, see https://lmod.readthedocs.io), use that term once and then mention we'll refer to it as "module system" for short